dali/kernels/context.h

```diff
@@ -146,23 +146,29 @@ class Scratchpad {
   if_iterable<Collection, T*>
   ToHost(const Collection &c) {
     T *ptr = AllocateHost<T>(dali::size(c));
-    std::copy(begin(c), end(c), ptr);
+    T *it = ptr;
+    for (const auto &x : c)
+      new(it++)T(x);
```

Comment:
Bug fix: the previous code didn't work with objects that had non-trivial copy assignment. It's still a big of an abuse to use such objects with scratchpad at all, but here we go :\